src/producer/base_producer.rs

```diff
@@ -362,19 +360,32 @@ where
     /// Regular calls to `poll` are required to process the events and execute
     /// the message delivery callbacks.
     pub fn poll<T: Into<Timeout>>(&self, timeout: T) {
-        let event = self.client().poll_event(&self.queue, timeout.into());
-        if let EventPollResult::Event(ev) = event {
-            let evtype = unsafe { rdsys::rd_kafka_event_type(ev.ptr()) };
-            match evtype {
-                rdsys::RD_KAFKA_EVENT_DR => self.handle_delivery_report_event(ev),
-                _ => {
-                    let evname = unsafe {
-                        let evname = rdsys::rd_kafka_event_name(ev.ptr());
-                        CStr::from_ptr(evname).to_string_lossy()
-                    };
-                    warn!("Ignored event '{}' on base producer poll", evname);
+        let mut remaining = if let Timeout::After(dur) = timeout.into() {
+            dur
+        } else {
+            Duration::MAX
+        };
+        let mut attempt = 0;
+        while attempt > 0 && remaining > Duration::ZERO {
+            let start = Instant::now();
```

Comment:
You can simplify code here by introducing simple utility type for measuring  time.

```
struct Deadline {
    deadline: Instant
}

impl Deadline {
    fn new(duration: Duration) -> Self {
         Self { deadline: Instant::now().add(...
    }

   fn remaining(&self) -> Duration {
      ...
   }
 
  fn elapsed() -> bool { 
      ....
  }
}
```

Reply:
Fixed in 36d4ebfa.